### ISSUE TITLE
3.4.2.3.2 Reading Order in EPUB A11Y 1.1

### ISSUE THREAD

**Issue by t-kudou** [mattgarrish/epub-revision#2222]
https://www.w3.org/TR/epub-a11y-11/#sec-mo-order

Objective
Ensure Media Overlay playback matches logical reading order.

It is a very welcome step forward in the handling of Japanese that the order of reading should be in logical order.
In the past, reading order by Media Overlays was bound by the rule of HTML order, which was not sufficient in many cases for handling Japanese.

Will pursuant to the provisions of this section allow us to use Media Overlays to reading table cells vertically, mathematical expressions, chemical formulas, Chinese sentences (Kanbun), etc. in SMIL in their original logical order, rather than in HTML order?

**Comment by murata2makoto** [comment]
I am not sure what "logical reading order"  means here.  Does it mean what sounds logical to the user?  Or, does it mean the depth-first order in the document tree?

**Comment by t-kudou** [comment]
I think a logical order for users is important from an accessibility standpoint.

**Comment by mattgarrish** [comment]
> Does it mean what sounds logical to the user? Or, does it mean the depth-first order in the document tree?

What is logical for the user. This is what the "understanding" section says:

> Ordering the playback to match the default reading order is the safest way to ensure that users can follow the text. In some cases, however, strict adherence to this practice could result in a suboptimal reading experience (e.g., playback of a table by column instead of row might make more logical sense in some cases).

We changed this a while back because it was brought up first with epubcheck that strictly following the DOM does not always lead to logical reading sequences in all languages.

This is a success criterion that takes human interpretation to evaluate, as a machine cannot check if the playback order makes logical sense.

But it might be better to switch "default reading order" with "DOM order" in the quote above to make extra clear what we're talking about.

**Comment by murata2makoto** [comment]
@mattgarrish 

Thanks.  I understand now.   Surprised to learn that the original restriction (the DOM order only) is lifted.

**Comment by clapierre** [comment]
my vote would be to include an additional statement to the reading order (ie.  "DOM order" )

e.g.
Ensure Media Overlay playback matches logical reading order (aka "DOM order).
or
Ensure Media Overlay playback matches DOM order (aka "logical reading order").

**Comment by mattgarrish** [comment]
> Ensure Media Overlay playback matches logical reading order (aka "DOM order).

These are actually opposites, @clapierre. The DOM order is what you get when, for example, an HTML document is read in. It's the tree of nodes that results from the ordering of the elements in the document.

That ordering is forced on authors by HTML and it doesn't always match how users actually read the rendered content. As mentioned above, tables can be read by column in Japan rather than by row in some cases. Making media overlays strictly follow the DOM prevents Japanese publishers from providing users a read-back presentation that matches what they would want.

**Comment by clapierre** [comment]
I think we need to be clear then on 
"logical reading order" vs. "default reading order" vs. "DOM order" 
because it is not clear yet to me exactly where the issue is and how we need to address this to make it clear to readers what they should be doing to allow for various reading systems to "do the right thing".

**Comment by mattgarrish** [comment]
Default reading order is generally understood in EPUB as the combination of spine order with DOM order. It's mechanistic.

Logical reading order is how users would read the content in their natural language, which can't always be represented at the markup level. Sometimes you have to put vertical reading into a horizontal layout.

Using the table case, an AT allows users to move down the column if that's the expected path, but media overlays would force all users to read across rows if you have to follow the dom without exception. Without a similar table reading mode, which I don't believe has been implemented, we're forcing users to read content in a less accessible manner.

But markup is better at matching horizontal reading patterns than vertical, which is why this problem doesn't often occur in Romance languages, for example.

**Comment by mattgarrish** [comment]
Here is the original issue that @rdeltour opened against the specification: https://github.com/w3c/epub-specs/issues/1458

It came about because of problems with the document order of paragraphs not being able to match the expected reading order.

The changes were implemented in: https://github.com/w3c/epub-specs/pull/1558

Keep in mind that it's still recommended to follow the DOM, but exceptions can be made. The reading order still has to make sense; we're not passing any media overlay with a nonsensical reading order. (Keeping in mind that these are not yet strict requirements to follow, either, so perhaps we are.)
